tests/components/test_wake_on_lan.py

```diff
@@ -0,0 +1,53 @@
+"""Tests for Wake On LAN component."""
+import asyncio
+from functools import partial
+from unittest import mock
+
+import pytest
+
+from homeassistant.setup import async_setup_component
+from homeassistant.components.wake_on_lan import (
+    DOMAIN, SERVICE_SEND_MAGIC_PACKET)
+
+
+@pytest.fixture
+def mock_wakeonlan():
+    """Mock mock_wakeonlan."""
+    module = mock.MagicMock()
+    with mock.patch.dict('sys.modules', {
+        'wakeonlan': module,
+    }):
+        yield module
+
+
+@asyncio.coroutine
+def test_setup_component(hass, mock_wakeonlan):
+    """Test the set up of new component."""
+    assert(not hass.services.has_service(DOMAIN, SERVICE_SEND_MAGIC_PACKET))
+    yield from async_setup_component(hass, DOMAIN, {})
+    assert(hass.services.has_service(DOMAIN, SERVICE_SEND_MAGIC_PACKET))
+
+
+@asyncio.coroutine
+def test_send_magic_packet(hass, caplog, mock_wakeonlan):
+    """Test of send magic packet service call."""
+    yield from async_setup_component(hass, DOMAIN, {})
+
+    yield from hass.async_add_job(partial(
```

Comment:
@balloob, here I had a problem that my short experience couldn't resolve:
When just yielding from `hass.services.call`, **the test never ends**, as if the call never executes. Only when adding the job the test runs ok and ends immediately. I'll try some more, but I will appreciate some light here...